### PR TITLE
[TIMOB-26550] Add support for file property to HTTPClient

### PR DESCRIPTION
**Pull request by infosia**
[TIMOB-26550](https://jira.appcelerator.org/browse/TIMOB-26550)

- Implement `HTTPClient.file` property.

```js
function createButton(title, filename, url) {
    var b = Ti.UI.createButton({
        title: title
    });

    b.addEventListener("click", function () {
        var xhr = Titanium.Network.createHTTPClient();
        if (filename != null) {
            xhr.file = filename;
        }
        xhr.onload = function () {
            var responseFilePath = this.responseData.nativePath;
            alert("responseData.nativePath = " + responseFilePath);
            if (responseFilePath != null) {
                imageView.image = responseFilePath;
            } else {
                imageView.image = this.responseData;
            }
        };
        xhr.open('GET', url);
        xhr.send();
    });

    return b;
}


var win = Titanium.UI.createWindow({
    title: "The 'file' property for HTTPClient",
    layout: "vertical"
});

var imageView = Titanium.UI.createImageView({
    top: 5,
    height: 200,
    width: 300
});

win.add(createButton("Not setting 'file'", null, "http://2.bp.blogspot.com/-z-cZ7JirojQ/Tc8OMAAqbLI/AAAAAAAAAfc/slhGE764cyY/s1600/Cars+2.jpg"));
win.add(createButton("'file' is in tempDirectory", Ti.Filesystem.tempDirectory + "/fruit.jpg", "http://southernbite.com/wp-content/uploads/2012/05/SouthernBiteFruitSalad-2.jpg"));
win.add(createButton("'file' is in applicationDataDirectory", Ti.Filesystem.applicationDataDirectory + "/fruit.jpg", "http://southernbite.com/wp-content/uploads/2012/05/SouthernBiteFruitSalad-2.jpg"));

win.add(imageView);

win.open();
```

Expected: 
- Pushing `Not setting 'file'` button should show `responseData.nativePath = null`
- Rest of buttons should show the path where the image is saved

### CODE REVIEW

File: Source/Network/src/HTTPClient.cpp [hunk 1]
```diff
@@ -247,6 +248,19 @@ namespace TitaniumWindows
 			}, CONTINUATION_CONTEXT())
 				.then([this](task<Windows::Storage::Streams::IBuffer^> previousTask) {
 				try {
+
+					if (!file__.empty()) {
+						auto Titanium = static_cast<JSObject>(get_context().get_global_object().GetProperty("Titanium"));
```
Comment (garymathews) [inline Source/Network/src/HTTPClient.cpp]: These can be `const` too?

File: Source/Network/src/HTTPClient.cpp [hunk 2]
```diff
@@ -251,6 +252,19 @@ namespace TitaniumWindows
 			}, CONTINUATION_CONTEXT())
 				.then([this](task<Windows::Storage::Streams::IBuffer^> previousTask) {
 				try {
+
+					if (!file__.empty()) {
+						const auto Titanium = static_cast<JSObject>(get_context().get_global_object().GetProperty("Titanium"));
+						const auto Filesystem = static_cast<JSObject>(Titanium.GetProperty("Filesystem"));
+						auto File = static_cast<JSObject>(Filesystem.GetProperty("File"));
```
Comment (garymathews) [inline Source/Network/src/HTTPClient.cpp]: `const auto` ?

File: Source/TitaniumKit/src/Network/HTTPClient.cpp [hunk 3]
```diff
@@ -347,11 +347,15 @@ namespace Titanium
 
 		TITANIUM_PROPERTY_GETTER(HTTPClient, responseData)
 		{
-			auto Blob = get_context().CreateObject(JSExport<Titanium::Blob>::Class());
+			auto Titanium = static_cast<JSObject>(get_context().get_global_object().GetProperty("Titanium"));
```
Comment (garymathews) [inline Source/TitaniumKit/src/Network/HTTPClient.cpp]: `const auto` here too?

Comment (infosia) [reply Source/TitaniumKit/src/Network/HTTPClient.cpp]: Yes fixed 👍

File: Source/Network/src/HTTPClient.cpp (hunk 2 above)
Comment (infosia) [reply Source/Network/src/HTTPClient.cpp]: No, this should not be `const `here because this needs to call `CallAsConstructor` later which is not const.

**Comment by keerthi1032** [comment]
FR Passed.works as expected.
  Name                        = Microsoft Windows 10 Pro
  Version                     = 10.0.17134
  Architecture                = 64bit
  Memory                      = 17034395648
Node.js
  Node.js Version             = 8.9.1
  npm Version                 = 5.5.1
Titanium CLI
  CLI Version                 = 5.1.1
Titanium SDK
  SDK Version                 = 8.1.0.v20190219213249 - Copy
  SDK Path                    = C:\ProgramData\Titanium\mobilesdk\win32\8.1.0.v20190219213249 - Copy
  Target Platform             = windows
EMulator =windows 10 emulator

**Pull request merged**